internal/datacoord/index_service.go

```diff
@@ -237,6 +240,44 @@ func (s *Server) CreateIndex(ctx context.Context, req *indexpb.CreateIndexReques
 	return merr.Success(), nil
 }
 
+func ValidateIndexParams(index *model.Index, key, value string) error {
+	switch key {
+	case common.MmapEnabledKey:
+		indexType := getIndexType(index.IndexParams)
+		if !indexparamcheck.IsMmapSupported(indexType) {
+			return merr.WrapErrParameterInvalidMsg("index type %s does not support mmap", indexType)
+		}
```

Comment:
make mmap enabled false for mmap not support index will also return error here. sounds a little bit tricky

Reply:
this means mmap feature is disabled for the given index type, so we should forbid all operations on the feature, no matter enabling or disabling the mmap